### PR TITLE
38 login component

### PR DESCRIPTION
**Pull request by alaa-yasin**

### CODE REVIEW

File: client/src/components/utils/Login/index.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+          headers: { 'Content-Type': 'application/json' },
+          body: JSON.stringify({
+            tableNumber,
+            secret,
+          }),
+        });
+      })
+      .then(res => {
+        if (res.status === 200) {
+          res.json();
```
Comment (ahmedisam99) [inline client/src/components/utils/Login/index.js]: no need for this line
```suggestion
          
```

File: client/src/components/utils/Login/index.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+          history.push('/');
```
Comment (ahmedisam99) [inline client/src/components/utils/Login/index.js]: Before redirecting to the home page update the state for the app component adding the table number to it. To be accessible for all components

File: client/src/components/utils/Login/index.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+          </Button>
+        </form>
+        <p className="login__err-message">{err}</p>
+      </div>
+    );
+  }
+}
+
+Login.propTypes = {
+  history: propTypes.string.isRequired,
```
Comment (ahmedisam99) [inline client/src/components/utils/Login/index.js]: `history` is not a string, hence `history.push()`!

File: client/src/components/utils/Login/index.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
+import React from 'react';
```
Comment (MohammadAlhalaq) [inline client/src/components/utils/Login/index.js]: I think login is not utils  :)

Comment (alaa-yasin) [reply client/src/components/utils/Login/index.js]: Yes, you're right. I'll create a new folder called "pages" and put all pages components inside it.

File: client/src/components/App/index.js [hunk 5]
```diff
@@ -1,14 +1,33 @@
...
+  };
+
+  render() {
+    return (
+      <Router>
+        <Switch>
+          <Route
+            exact
+            path="/login"
+            component={props => (
```
Comment (ahmedisam99) [inline client/src/components/App/index.js]: If you're using an inline function for rendering use `render`
[react-router-dom `Route/component`](https://reacttraining.com/react-router/web/api/Route/component)
[react-router-dom `Route/render`](https://reacttraining.com/react-router/web/api/Route/render-func)
```suggestion
            render={props => (
```

File: client/src/components/App/index.js [hunk 6]
```diff
@@ -1,14 +1,33 @@
...
-      </Switch>
-    </Router>
-  );
-};
+export default class App extends React.Component {
+  state = {
+    tableNumber: null,
+  };
+
+  updateTableNumber = e => {
```
Comment (ahmedisam99) [inline client/src/components/App/index.js]:
```suggestion
  updateTableNumber = tableNumber => {
```
Use declerative args names

File: client/src/components/utils/Login/index.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+          headers: { 'Content-Type': 'application/json' },
+          body: JSON.stringify({
+            tableNumber,
+            secret,
+          }),
+        });
+      })
+      .then(res => {
+        if (res.status === 200) {
+          updateTableNumber(tableNumber);
```
Comment (ahmedisam99) [inline client/src/components/utils/Login/index.js]: `updateTableNumber` calls `setState` which means the process of updating the table number in the `App` component is `async` which may cause a problem when there is trouble updating the table number in the `App` component as you technically redirecting the user before updating the state value in the app.

### Solution
refactor `updateStateNumber` to take a callback and call `history.push` in the callback.

**Pull request merged**